handler/handler.go

```diff
@@ -27,7 +27,14 @@ func AddTask(w http.ResponseWriter, r *http.Request) {
 	err = json.Unmarshal(body, &request)
 	handleError(err, w)
 
-	createRequest(request, w)
+	taskId, err := scheduleTask(scheduler, request)
+	if err != nil {
+		writeJSON(500, err, w)
+		return
+	}
+
+	w.Header().Set("Location", fmt.Sprintf("/task/%s", taskId))
```

Comment:
Was reading up, and seems http.Request has the Host variable that is supposedly the full hostname, so could build an absolute path.